src/solver/libModelObject/model.cpp

```diff
@@ -19,14 +19,120 @@
 ** along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
 */
 
+#include <algorithm>
+#include <iterator>
+#include <utility>
+#include <vector>
+
 #include <antares/solver/libObjectModel/model.h>
 
 namespace Antares::Solver::ObjectModel
 {
 
-std::vector<Constraint*> getConstraints()
+/**
+ * \brief Builds and returns the Model object.
+ *
+ * \return The constructed Model object.
+ */
+Model ModelBuilder::build()
```

Comment:
Main question here (already asked about **modelConverter.cpp**) : why separate construction from the rest of the class **Model** (mainly getters) by using a model builder ?

Reply:
https://github.com/AntaresSimulatorTeam/Antares_Simulator/pull/2431#discussion_r1787284646